specification/streamanalytics/resource-manager/Microsoft.StreamAnalytics/stable/2020-03-01/inputs.json

```diff
@@ -0,0 +1,1084 @@
+        }
+      },
+      "required": [
+        "type"
+      ]
+    },
+    "BlobReferenceInputDataSource": {
+      "description": "Describes a blob input data source that contains reference data.",
+      "x-ms-discriminator-value": "Microsoft.Storage/Blob",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ReferenceInputDataSource"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/BlobReferenceInputDataSourceProperties",
+          "description": "The properties that are associated with a blob input containing reference data. Required on PUT (CreateOrReplace) requests.",
+          "x-ms-client-flatten": true
+        }
+      }
+    },
+    "BlobReferenceInputDataSourceProperties": {
+      "description": "The properties that are associated with a blob input containing reference data.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/BlobDataSourceProperties"
+        }
+      ],
+      "properties": {}
+    },
+    "BlobDataSourceProperties": {
+      "description": "The properties that are associated with a blob data source.",
+      "properties": {
+        "storageAccounts": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageAccount"
+          },
+          "description": "A list of one or more Azure Storage accounts. Required on PUT (CreateOrReplace) requests."
+        },
+        "container": {
+          "type": "string",
+          "description": "The name of a container within the associated Storage account. This container contains either the blob(s) to be read from or written to. Required on PUT (CreateOrReplace) requests."
+        },
+        "pathPattern": {
+          "type": "string",
+          "description": "The blob path pattern. Not a regular expression. It represents a pattern against which blob names will be matched to determine whether or not they should be included as input or output to the job. See https://docs.microsoft.com/en-us/rest/api/streamanalytics/stream-analytics-input or https://docs.microsoft.com/en-us/rest/api/streamanalytics/stream-analytics-output for a more detailed explanation and example."
+        },
+        "dateFormat": {
+          "type": "string",
+          "description": "The date format. Wherever {date} appears in pathPattern, the value of this property is used as the date format instead."
+        },
+        "timeFormat": {
+          "type": "string",
+          "description": "The time format. Wherever {time} appears in pathPattern, the value of this property is used as the time format instead."
+        }
+      }
+    },
+    "StorageAccount": {
+      "description": "The properties that are associated with an Azure Storage account",
+      "properties": {
+        "accountName": {
+          "type": "string",
+          "description": "The name of the Azure Storage account. Required on PUT (CreateOrReplace) requests."
+        },
+        "accountKey": {
```

Comment:
this needs to be marked as a secret and ensure it doesn't get returned in responses